src/glaciers/glacier/glacier2D_utils.jl

```diff
@@ -646,4 +659,76 @@ function is_in_glacier(A::Matrix{F}, distance::I) where {I <: Integer, F <: Abst
         B .= min.(B, circshift(B, (1,0)), circshift(B, (-1,0)), circshift(B, (0,1)), circshift(B, (0,-1)))
     end
     return B .> 0.001
-end
\ No newline at end of file
+end
+
+"""
+    block_average_pad_edge(mat::Matrix{F}, n::Int) where {F <: AbstractFloat}
+
+Downsamples a matrix by averaging `n x n` blocks, using edge-replication padding
+when the matrix dimensions are not divisible by `n`.
+Edge padding replicates the last row/column values to expand the matrix so that both
+dimensions are divisible by `n`.
+Returns a matrix of averaged values with size `(ceil(Int, X/n), ceil(Int, Y/n))`.
+
+Arguments
+- `mat::Matrix{F}`: Input 2D matrix.
+- `n::Int`: Block size for downsampling.
+"""
+function block_average_pad_edge(mat::Matrix{F}, n::Int) where {F <: AbstractFloat}
+    X, Y = size(mat)
+    new_X = ceil(Int, X / n) * n
+    new_Y = ceil(Int, Y / n) * n
+
+    # Create padded matrix filled with edge values
+    padded = similar(mat, new_X, new_Y)
+
+    # Fill original data
+    padded[1:X, 1:Y] .= mat
+
+    # Pad bottom rows with last row
+    if new_X > X
+        for i in X+1:new_X
+            padded[i, 1:Y] .= mat[end, :]
+        end
+    end
+
+    # Pad right columns with last column
+    if new_Y > Y
+        for j in Y+1:new_Y
+            padded[1:X, j] .= mat[:, end]
+        end
+    end
+
+    # Fill bottom-right corner (if both X and Y were padded)
+    if new_X > X && new_Y > Y
+        for i in X+1:new_X
+            for j in Y+1:new_Y
+                padded[i, j] = mat[end, end]
+            end
+        end
+    end
+
+    return block_average(padded, n)
+end
+
+"""
+    block_average(mat::Matrix{F}, n::Int) where {F <: AbstractFloat}
+
+Downsamples a matrix by averaging non-overlapping `n x n` blocks.
+Returns a matrix of the block-averaged values with size `(div(X, n), div(Y, n))`
+where `(X, Y) = size(mat)`.
+
+Arguments
+- `mat::Matrix{F}`: Input 2D matrix.
+- `n::Int`: Block size for downsampling. Both matrix dimensions must be divisible by `n`.
+"""
+function block_average(mat::Matrix{F}, n::Int) where {F <: AbstractFloat}
+    X, Y = size(mat)
+    @assert X % n == 0 && Y % n == 0 "Matrix dimensions are $(size(mat)) but they are not divisible by n=$n"
```

Comment:
@albangossard why do you need this here? I noticed you don't have this assess in `block_average_pad_edge`. Do you think this can be restrictive for some test cases (e.g., there is no integer larger than 1 that divides both X and Y)?